_source/_posts/2022-02-28-spring-native-jhipster.adoc

```diff
@@ -0,0 +1,451 @@
+
+////
+*Prerequisites*:
+
+- https://sdkman.io/[Java 17 with GraalVM]+
+- https://docs.docker.com/desktop/#download-and-install[Docker Desktop]
+////
+
+toc::[]
+
+== Why should you care about serverless?
+
+From https://www.ibm.com/cloud/learn/serverless[IBM's What is Serverless computing?]
+
+> Serverless is a cloud execution model that enables a simpler, more cost-effective way to build and operate cloud-native applications.
+
+That's it!
+
+For companies with high traffic and large cloud bills, serverless makes sense. They can save millions of dollars a month.
+
+== GraalVM allows serverless Java for everyone!
+
+I've been playing with GraalVM and the Java frameworks that support it for around 18 months. After doing a lot of research, I blogged about link:/blog/2021/06/18/native-java-framework-comparison[building native Java apps with Micronaut, Quarkus, and Spring Boot].
+
+In September 2021, I did a talk with Josh Long about Spring Native with JHipster at the San Francisco JUG. I flew out on a Monday, we got everything working by Wednesday afternoon, and then spoke about our experience on Wednesday night. I wrote about it in https://www.linkedin.com/pulse/jhipster-works-spring-native-matt-raible/[JHipster Works with Spring Native!] on LinkedIn and created a https://github.com/jhipster/generator-jhipster/issues/16498[JHipster issue] to automate our learnings.
+
+++++
+{% twitter 1442698739563446276 %}
+++++
+
+== What is Spring Native?
+
+Spring Native provides an API to configure Spring Boot and GraalVM so classes that are not easily discoverable become recognizable. For example, those that are instantiated using reflection. It's a really slick extension to Spring Boot and will likely disappear with Spring Boot 3 because it'll be native by default.
+
+As a result of Josh and I's successful https://youtu.be/F9oydL_MndA[presentation at the SF JUG], we were invited to speak at the Garden State JUG in December. We laid low for a few months and rekindled our research in December. I wrote about it in https://www.linkedin.com/pulse/jhipster-works-spring-native-part-2-matt-raible/[JHipster works with Spring Native, Part 2!]
+
+== 🔥 Announcing the JHipster Native blueprint v1.0!
+
+In late January 2022, we (the JHipster team) upgraded JHipster's to use Spring Boot 2.6 and released version https://www.jhipster.tech/2022/01/23/jhipster-release-7.6.0.html[7.6.0].
+
+In early February, I updated the Spring Native with JHipster examples that Josh and I'd been using for research. I was ready to start automating the Spring Native integration using a JHipster module. When https://github.com/jhipster/generator-jhipster/issues/16498#issuecomment-1030263905[I asked the JHipster team] about the best way to implement it, https://github.com/mshima[Marcelo Shima] volunteered to create the initial blueprint.
+
+This was on a Friday afternoon (mountain time). By the next morning, Marcelo had the MVP finished and provided steps to reproduce our examples. 😳
```

Comment:
```suggestion
This was on a Friday afternoon (Mountain Time or MT). By the next morning, Marcelo had the MVP finished and provided steps to reproduce our examples. 😳
```
upper case for the time zone

Reply:
Do we care about MDT or MST? I think MT is good enough, so I'll commit this.